src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -206,7 +206,8 @@ void FFN<
     const size_t effectiveBatchSize = std::min(batchSize,
         size_t(predictors.n_cols) - i);
 
-    MatType predictorAlias(predictors.colptr(i), predictors.n_rows,
+    const MatType predictorAlias(
+        (typename MatType::elem_type*) predictors.colptr(i), predictors.n_rows,
```

Comment:
Suggestion: instead of plain C style cast, perhaps use `const_cast` to explicitly indicate the intent.

Alternatively, `predictors.unsafe_col(i)` might be another way to create the alias.